easybuild/tools/options.py

```diff
@@ -634,6 +634,7 @@ def job_options(self):
         opts = OrderedDict({
             'backend-config': ("Configuration file for job backend", None, 'store', None),
             'cores': ("Number of cores to request per job", 'int', 'store', None),
+            'max-jobs': ("Maximum number of concurrent jobs (queued or running)", 'int', 'store', 50),
```

Comment:
@riccardomurri Shouldn't we set to default to `0` (which I'm guessing means no limit), to maintain current behaviour?

In any case, `50` seems so arbitrary...

Thoughts @victorusu?